src/pages/Homepage.jsx

```diff
@@ -1,70 +1,83 @@
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
+import { v4 as uuid } from 'uuid';
 import TimerComponent from '../components/TimerComponent';
 import ActiveTeamsComponent from '../components/ActiveTeamsComponent';
 import EliminatedTeamsComponent from '../components/EliminatedTeamComponent';
+import findTeam from '../utils/index.js';
 
 const Homepage = () => {
   const [teams, setTeams] = useState([]);
   const [draggedIndex, setDraggedIndex] = useState(null);
 
   const handleAddTeam = (type) => {
-    setTeams([
-      ...teams,
+    setTeams((prevState) => [
```

Comment:
If you're setting a new state by refering to the old state always use the callback function. Otherwise it leads to data inconsistancies in bigger applications.

Reply:
Oh, thanks @iteratus  for correcting me here. I so much appreciate it. Noted.